File: algorithms/1893.check-if-all-the-integers-in-a-range-are-covered/solution.go
func isCovered(ranges [][]int, left int, right int) bool {
	covered := make([]bool, 51)

	for _, v := range ranges {
		for i := v[0]; i <= v[1]; i++ {
			covered[i] = true
		}
	}

	for i := left; i <= right; i++ {
		if !covered[i] {
			return false
		}
	}

	return true
}

File: algorithms/1893.check-if-all-the-integers-in-a-range-are-covered/solution.rs
impl Solution {
    pub fn is_covered(ranges: Vec<Vec<i32>>, left: i32, right: i32) -> bool {
        let mut covered = vec![false; 51];

        for range in ranges {
            for i in range[0]..=range[1] {
                covered[i as usize] = true;
            }
        }

        for i in left..=right {
            if !covered[i as usize] {
                return false;
            }
        }

        true
    }
}
